fastparquet/core.py

```diff
@@ -249,7 +249,7 @@ def read_col(column, schema_helper, infile, use_cat=False,
             null = not schema_helper.is_required(cmd.path_in_schema[0])
             null_val = (se.repetition_type !=
                         parquet_thrift.FieldRepetitionType.REQUIRED)
-            num = encoding._assemble_objects(assign, defi, rep, val, dic, d,
```

Comment:
So the output of `_assemble_objects` should indeed be the number of rows, but L294 has another `num +=` based on the number of *values*, which is why, presumably, it looks we are done prematurely.